Examples/Scripts/MaterialMapping/GeometryVisualisationAndMaterialHandling.py

```diff
@@ -0,0 +1,784 @@
+
+            z_shift = 0.
+            if entry['value']['transform']['translation'] != None:
+                z_shift = entry['value']['transform']['translation'][2]
+
+            if entry['value']['type'] == "CylinderSurface":
+                extends = [0., entry['value']['bounds']['values'][0],
+                           z_shift-entry['value']['bounds']['values'][1], z_shift+entry['value']['bounds']['values'][1]]
+            else:
+                extends = [entry['value']['bounds']['values'][0], entry['value']['bounds']['values'][1],z_shift]
+
+            if "approach" in entry:
+                extends.append(entry["approach"])
+            else:
+                extends.append(-1)
+            extends.append(entry["layer"])
+
+            if entry['value']['type'] == "CylinderSurface":
+                if "approach" in entry:
+                    index_to_extends_layers_approach_cylinders[vol-1].append(extends)
+                else:
+                    index_to_extends_layers_cylinders[vol-1].append(extends)
+            else:
+                if "approach" in entry:
+                    index_to_extends_layers_approach_discs[vol-1].append(extends)
+                else:
+                    index_to_extends_layers_discs[vol-1].append(extends)
+
+        if "boundary" in entry:
+            extends = []
+            vol = entry['volume']
+
+            z_shift = 0.
+            if entry['value']['transform']['translation'] != None:
+                z_shift = entry['value']['transform']['translation'][2]
+
+            if entry['value']['type'] == "CylinderSurface":
+                extends = [0., entry['value']['bounds']['values'][0],
+                           z_shift-entry['value']['bounds']['values'][1], z_shift+entry['value']['bounds']['values'][1],entry['boundary']]
+                index_to_extends_layers_bounds_cylinders[vol-1].append(extends)
+            else:
+                extends = [entry['value']['bounds']['values'][0], entry['value']['bounds']['values'][1],z_shift,entry['boundary']]
+                index_to_extends_layers_bounds_discs[vol-1].append(extends)
+
+    # Steering the information and collect it into an output file if needed
+
+    interesting_volumes = []
+    v_index = 0
+    for elements in index_to_extends_layers_cylinders:
+        for coords in elements:
+            if v_index not in interesting_volumes:
+                interesting_volumes.append(v_index)
+            if index_to_names[v_index] not in steering_map:
+                steering_map[index_to_names[v_index]] = index_info(v_index+1)
+            steering_map[index_to_names[v_index]].layers_no_approach.append(coords[5])
+        v_index = v_index+1
+
+    v_index = 0
+    for elements in index_to_extends_layers_discs:
+        for coords in elements:
+            if v_index not in interesting_volumes:
+                interesting_volumes.append(v_index)
+            if index_to_names[v_index] not in steering_map:
```

Comment:
The logic here is straightforward. However, due to depth of the structure not super straightforward to capture. 
Could a helper function alike `appendIfMissing(dict, index)` help?